packages/terminal/src/widget.ts

```diff
@@ -256,11 +274,15 @@ export class Terminal extends Widget {
     switch (msg.type) {
       case 'stdout':
         if (msg.content) {
-          this._term.write(msg.content[0] as string);
+          this._ready.promise.then(() => {
```

Comment:
Is there a danger of outputting messages in the wrong order here, now that it is asynchronous? I wonder if it would be better to await the terminal being ready before hooking up the message handler (in the `session` setter above).

Reply:
good idea.

Reply:
moved and in turn took ready out of a number of private messages... i think all the public/protected methods need to keep them, though...

Reply:
Sounds good, thanks!